processing/src/main/java/io/druid/segment/IndexMaker.java

```diff
@@ -1600,6 +1602,10 @@ public Integer apply(Integer input)
       this.convertedDims = convertedDims;
       this.converters = converters;
       this.indexNumber = indexNumber;
+      this.converterArray = new IntBuffer[convertedDims.size()];
+      for(int i = 0; i < convertedDims.size(); ++i){
```

Comment:
I would prefer if the initialization was done lazily when requesting the iterator, rather than at construction time to avoid reading data until it's actually needed.